### ISSUE TITLE
Bug: `context.cwd` is incorrect set as `/` when absolute path is used as `filename` in rule tester

### ISSUE THREAD

**Issue by JounQin** [typescript-eslint/typescript-eslint#11274]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

rule-tester

### Repro Code

```TypeScript
https://github.com/typescript-eslint/typescript-eslint/blob/936f35022c1e1357da82c4b958b7bff2563e2075/packages/rule-tester/src/RuleTester.ts#L215-L217

https://github.com/typescript-eslint/typescript-eslint/blob/936f35022c1e1357da82c4b958b7bff2563e2075/packages/rule-tester/src/RuleTester.ts#L225
```

### ESLint Config

```javascript

```

### tsconfig

```jsonc

```

### Expected Result

`context.cwd` should not be set `/`

### Actual Result

`context.cwd === '/'`

### Additional Info

I thought it was ESLint's issue #19821, and found out it's actually `typescript-eslint`'s.

### Versions

| package                                | version |
| `@typescript-eslint/rule-tester`       | `8.33.0` |

**Comment by nayounsang** [comment]
Oh, okay. I found the problematic area and it's reproduced.
For ex)
```ts
    it('passes with correct array output', () => {
      expect(() => {
        ruleTester.run('my-rule', rule, {
          invalid: [
            {
              code: 'foo',
              errors: [{ messageId: 'error' }],
              filename: '/test/file.ts',
              output: ['bar', 'baz'],
            },
          ],
          valid: [],
        });
      }).not.toThrow();
    });
```
In this case, the "/" path really comes out!
There are many other test cases. Analyze them more and add appropriate test cases & fix bugs.

**Comment by reduckted** [comment]
@JounQin can you describe the expected result? It's not clear what you expect from your description.

**Comment by JounQin** [comment]
@reduckted 

According to https://github.com/eslint/eslint/issues/19821#issuecomment-2944904637

> `context.cwd` should, I believe, always be `process.cwd()` when rules are run through RuleTester.

**Comment by nayounsang** [comment]
I don't quite understand why it has to be process.cwd(). In fact, some tests fail.
Let's say process.cwd() is /A/B/C. Here's an example of giving a filename and what the path in the context is expected to be.
1. ab path
given: `/foo.ts`
expect: `/foo.ts`

2. re path
given: `../foo.ts`
expect: `A/B/foo.ts`

3. just filename
given: `foo.ts`
expect: `A/B/C/foo.ts` (Internally processed with process.cwd())

Wouldn't this be accurate?
If set it to process.cwd(),
```ts
// RuleTester.#testInvalidTemplate
    const result = this.runRuleForItem(ruleName, rule, item);
    const messages = result.messages.map(message => ({
      ...message,
      ...(message.ruleId && {
        ruleId: message.ruleId.startsWith(RULE_TESTER_PLUGIN_PREFIX)
          ? message.ruleId.slice(RULE_TESTER_PLUGIN_PREFIX.length)
          : message.ruleId,
      }),
    }));
```
`result.messages.map(m=>m.ruleId)` is `[null]`, so test is failed. 
Further analysis is needed to determine why this is the case.

**Comment by JounQin** [comment]
> I don't quite understand why it has to be `process.cwd()`

I believe that's how ESLint actually works?

**Comment by kirkwaiblinger** [comment]
Genuine question - why would `context.cwd` exist at all if it's meant to be always equal to `process.cwd()`? Is there some kind of documentation around this? Right now the main two pieces of information I get out of the conversation in this issue are

- An eslint core maintainer has said that they _believe_ that `context.cwd` _should_ be equal to `process.cwd()` in the _eslint core_ Rule Tester. It's not clear to me whether that's a normative (how it ought to be) or descriptive (simply how it is) statement.
- The `context.cwd` was set the way it is in the _typescript-eslint_ Rule Tester as a pragmatic measure in order to address bugs with using file names work during test runs. It's not clear to me for now whether we know definitively that it either can or cannot be changed to be equal to `process.cwd()` without introducing regressions.

@JounQin Could you help us understand a _normative_ justification for the behavior you'd like to see? Thanks!

**Comment by JounQin** [comment]
@kirkwaiblinger It should always be `process.cwd()` **in rule tester**, not in actually with ESLint cli, which could be different, for example running ESLint in mono repo.

The key issue here is `context.cwd === '/'` for absolute filename.

**Comment by reduckted** [comment]
@JounQin Out of interest, what are you using `context.cwd` for? And what problem are you having where the "wrong" value in the rule tester is causing you trouble?

**Comment by JounQin** [comment]
@reduckted We're using `context.cwd` to resolve `glob` related paths, see also https://github.com/un-ts/eslint-plugin-import-x/pull/371, due to the current unexpected `context.cwd`, we can not remove the unnecessary fallback at https://github.com/un-ts/eslint-plugin-import-x/pull/371/files#diff-d26702cd663d2b4f4c73f51326b7daffe5327f32fe5628f99a86b874fe191dfeR347.

**Comment by reduckted** [comment]
Seems like what you really want to be able to do is _explicitly_ specify the `cwd` that the rule tester uses instead of having it set _implicitly_ based on the base path.

**Comment by JounQin** [comment]
> Seems like what you really want to be able to do is _explicitly_ specify the `cwd` that the rule tester uses instead of having it set _implicitly_ based on the base path.

Yes and no, what I want is how ESLint will be actually run, not only in rule tester.

**Comment by reduckted** [comment]
`cwd` can be set when you run ESLint, so being able to set that in the rule tester seems like exactly what you need.

**Comment by JounQin** [comment]
> `cwd` can be set when you run ESLint, so being able to set that in the rule tester seems like exactly what you need.

While that could work, but the default behavior should match no `cwd` set IMO.
